leshan-core/src/test/java/org/eclipse/leshan/core/node/codec/LwM2mNodeDecoderTest.java

```diff
@@ -83,6 +84,14 @@ public static void loadModel() {
         resForObj66.add(new ResourceModel(2, "res2", Operations.R, false, false, Type.OBJLNK, null, null, null));
         objects.add(
                 new ObjectModel(66, "object link tests 66", "", ObjectModel.DEFAULT_VERSION, true, false, resForObj66));
+        
+        // add object 25 from the LWM2M gateway specification
+        List<ResourceModel> resForObj25 = new ArrayList<>();
+        resForObj25.add(new ResourceModel(0, "Device ID", Operations.R, true, false, Type.STRING, null, null, null));
+        resForObj25.add(new ResourceModel(1, "Prefix", Operations.R, false, false, Type.STRING, null, null, null));
+        resForObj25.add(new ResourceModel(3, "IoT Device Objects", Operations.R, false, false, Type.CORELINK, null, null, null));
+        objects.add(
+                new ObjectModel(25, "Gateway", "", ObjectModel.DEFAULT_VERSION, true, false, resForObj25));
```

Comment:
I would prefer we use the object [3342.xml](https://github.com/OpenMobileAlliance/lwm2m-registry/blob/prod/3342.xml)

It is currently not used because it was added recently but we should use it in our tests.

Let's this aside for now, I will think a bit more how to make it available for test more easily.

Reply:
> There is lot of formatting issue but this is common issue from contribution where eclipse is not used 

And I feed vscode with the eclipse formatter parameters.. since it's supposed to be the same code under the hood..

> (This is probably a topic to address but hard to put it in priority list)

Or just ignore it and review with whitespace diff off :hand_over_mouth: 

> If I didn't missed something for now you only target TEXT and TLV correct ? any reason to not support other content format ?

Yes for now I want to make TEXT and TLV working and see for the other content type in a second step

> You should also add coreLink support to LwM2mTestObject class So you can easily test it with leshan-client-demo and leshan-server-demo.

Does mean we need to publish a new revision of the object @ OMA?

Reply:
> Does mean we need to publish a new revision of the object @ OMA?

Oops my mistake, it's not 3342 but  [3442](https://github.com/OpenMobileAlliance/lwm2m-registry/blob/prod/3442.xml), and it already contains a coreLink resource (single and multi instance)